### DISCUSSION TITLE
Use laminas/symfony components

### DISCUSSION THREAD

**Discussion by sreichel** [OpenMage/magento-lts#2932] [Ideas]
Switching to ZF1F is certainly a big step forward, but why not go one step further?

Some parts could be replaced by Laminas (ZF successor).

Shell commands could use Symfony console. (and we could integrate N98-magerun)

@cmuench @ktomk do you have an opinion on magerun part?

**Comment by cmuench** [comment]
@sreichel what's the idea? Include n98-magerun in the source of OpenMage?

**Reply by cmuench** [reply]
You can do that (license allow that) but some commands are designed to work without a installed system.

**Reply by sreichel** [reply]
This was just a spontaneous idea without giving it much thought.

Until it comes i prefer to help to make N98-magerun compatible to php8 :heart:

**Reply by sreichel** [reply]
@cmuench i dont know how much effort you (want) spent to M1-version ... maybe we have more progress when its part of OM? This was the reason i asked for ...

**Reply by cmuench** [reply]
My main focus will be on M2/Mage-OS.
If we integrate n98-magerun into OpenMage, then we can set the n98-magerun repository to read only.
On the other hand the tool is designed to work with multiple Magento instances. If we integrate into one instance then we loose the advantage. That's btw the idea behind the phar file.

Another idea. We bundle the phar file via `n98/magerun1-dist` and give the OpenMage Team the contribution permissions in the n98-magerun repository to help to refactor the code?

**Reply by fballiano** [reply]
At the moment I think magerun (which is just amazing) should stay independent, but in 1-2 years who knows :-)

**Reply by sreichel** [reply]
> We bundle the phar file via n98/magerun1-dist

I'm going crazy. In all these years i didn't get that repo exists :rofl: Great idea to bundle, so _we_ just have to finish php8 support. And in all these years I'd never to "complain" about reponse times :+1:  I'm not sure about  if its necessary to grant permerssion.

**Comment by addison74** [comment]
I did not allow myself to dream of such a thing. It would mean that OpenMage ensures a long future.

**Reply by sreichel** [reply]
What should i do when we finish php8 support .... ? :smile:

**Reply by addison74** [reply]
- Remove all old JS libraries and replace them with better alternatives
- Varnish support for large stores (Turpentine module or something else)
- A better catalog search (Elasticsearch, Sphinx, Sorl), which offers features like "Did you mean?" and anticipate the words.
- A better looking Backend theme based on Bootstrap or Tailwind/AlpineJS.
- A lot of basic missing features (editing an order, grids for product price and stock alerts, GDPR out of the box, better cookie management)
- ZF1-Future replaced by Laminas.

**Reply by sreichel** [reply]
Want to hire me? :rofl: 

What about to pick-up ...?

- https://github.com/nexcess/magento-turpentine
- https://github.com/Smile-SA/smile-magento-elasticsearch
- https://github.com/firegento/firegento-magesetup (GDRP)

> A better looking Backend theme 

Whats wrong with it? I'm happy with the new theme.

**Comment by ktomk** [comment]
I think it is a good idea to have this as one strategy for magento-lts. It would also make sense, as a very large part of the Magento 1 source code inherits from the ZF1 licensedcode technically, and IIRC Laminas does so as well, so this would be an upgrade.

No clue how much would break within Magerun-1 by that though, but given the perspective and my recent fiddling with the build there, it could make sense to branch in Magerun-1 in case it does and have an lts version that works with OpenMage Magento-LTS. At least an idea. OpenMage is one of the only versions that works on newer build systems IIRC.

But IMHO this should be first of all a question for OpenMage itself, not Magerun-1. Especially if I read your full question right, it means that not yet the ZF1 Framework has been excarved from OpenMage (no clue what the status quo is honestly), so it is yet unknown how switching the Magento1 core (the parenting framework it is written with) would work in general. From my subjective developers perspective I'd do that first to already have some results at hand which can only help with further decisions. Just my 2 cents.

**Reply by fballiano** [reply]
ZF1 is not in our repo anymore (although this version has not been released yet) but it's anyway strictly dependant on ZF1 code.

**Comment by fballiano** [comment]
Laminas is obviously great but:
- they're kinda strict about php version support, nowadays their composer.json states they only support php 8.0, 8.1 and 8.2, that would have a deep effect on us (it may be fine but that's different from what we've always done)
- starting with ZF2 the fetchAll/fetchCol/fetchPair methods were dropped in favour of the ResultSet pattern which I never loved, I guess that's the price for modernity
- that would really destroy compatibility with every extension that uses a Zend_Db_Expr or something like that. Now I think we shouldn't do it but in a couple of years, when php9 (or something like that) will make any un-updated extension useless anyway... then we can do whatever we want

:-)

**Reply by sreichel** [reply]
Require php8 should not block it.

In my opinion we can already set required version to 7.4, but that another story.

For laminas ...i dont want to replace ZF1 - just use components that are useful. Captcha, Mail, Json, ...? There is only one thing i'd like to get removed ... it's Zend_Pdf_, but therefor is no laminas replacement. (dompdf/tcpdf would be nice to have ...)
